### PR TITLE
Add a benchmark for dynamic column creation on partitioned tables

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

File: py/specs/dynamic_columns_on_partitioned_table.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+class statements:
+
+    def __init__(self):
+        self.names = column_names()
+
+    def __call__(self):
+        column_name = next(self.names)
+        return f'insert into t (p, {column_name}) values (?, ?)'
+
+    def __repr__(self):
```
Comment (matriv) [inline py/specs/dynamic_columns_on_partitioned_table.py]: why is this needed?

Comment (mfussenegger) [inline py/specs/dynamic_columns_on_partitioned_table.py]: So that run-spec and ./compare print something sensible.. otherwise it would be displayed as `<function .. at ...>` or something like that
